### PR TITLE
Openstack public vs private cloud slider

### LINKED ISSUE #9524: Add slider for representation of cost of public and private cloud
At `/openstack` build slider for graph to represent the difference in prices between public and private cloud. See the [comment](https://docs.google.com/document/d/1eqHK0hc9v7Mn-XV36QRyNkJenTXgBQR_-RULKm5rZ5o/edit?disco=AAAAIAiqPGY) in the copy doc.
- The TCO bar of public cloud should be lower (therefore green) than private cloud (red at this point) at first.
- As the VM hours increase, the private cloud TCO decreases and drops lower than public cloud and the colours of the bars swap. 

@waynecrosby has kindly offered to create a GIF to show how the slider should work. Here is a static image to show the graph:

image

### PR DESCRIPTION
**Pull request by carkod**
## Done

- Added slider for representation of cost of public and private cloud

## QA

- Check slider https://ubuntu-com-9917.demos.haus/openstack

See design in the issue

image

## Issue / Card

Fixes #9524

### CODE REVIEW

**Comment by anasereijo** [comment]
nice work @carkod 🥇 

Just one tiny thing:

when TCO hits the max it gets too close to the tile. Can you give a bit more space?

Screenshot 2021-06-18 at 12 10 27


thanks

**Comment by tytus-kurek** [comment]
@carkod thank you for the update.

Some comments below:
- Please decrease the public cloud CapEx to almost 0 (not 0, but close to 0).
- Please increase the public cloud OpEx to almost 0 when at the very left of the slider.
- Would it be possible to design it so that the private cloud TCO becomes lower than the public cloud TCO in the middle of the slider?

Again, have a look at the following source as a reference (slides 9 and 10):

https://docs.google.com/presentation/d/10ONsEguzeqnUq4dLkbeR9iKXedbXJXpvVcH4OuORCFc/edit#slide=id.gc34e1ef9b5_0_940

**Comment by carkod** [comment]
> nice work @carkod 
> 
> Just one tiny thing:
> 
> when TCO hits the max it gets too close to the tile. Can you give a bit more space?
> 
> Screenshot 2021-06-18 at 12 10 27
> 
> thanks

So I reduced the size of the cloud and used `h4` for the titles (Public Cloud and Private Cloud), so it resembles more the design (the text is larger and wider than the cloud images).

Also the comments by Tytus means that the bars now will never reach 100% total height so it will never touch the top.

**Comment by carkod** [comment]
> @carkod thank you for the update.
> 
> Some comments below:
> 
> * Please decrease the public cloud CapEx to almost 0 (not 0, but close to 0).
> * Please increase the public cloud OpEx to almost 0 when at the very left of the slider.
> * Would it be possible to design it so that the private cloud TCO becomes lower than the public cloud TCO in the middle of the slider?
> 
> Again, have a look at the following source as a reference (slides 9 and 10):
> 
> https://docs.google.com/presentation/d/10ONsEguzeqnUq4dLkbeR9iKXedbXJXpvVcH4OuORCFc/edit#slide=id.gc34e1ef9b5_0_940

- I put CapEx as value 3 and Opex value 1 (when slider is at the left tip), does this look ok? I think giving it less than 3 it will make it difficult to see the difference if there is only 1 px difference.
- I tried my best, we are not using discreet numbers, because we are using a scale property, but hopefully it's almost exactly in the middle now.

**Review by tytus-kurek — APPROVED** [review]
LGTM

**Comment by anasereijo** [comment]
lgtm 👍

File: templates/openstack/index.html [hunk 1]
```diff
@@ -724,12 +850,15 @@ <h2>Get in touch</h2>
     </div>
   </section>
 
+
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/6.5.0/d3.min.js"></script>
+  <script src="https://cdn.jsdelivr.net/npm/d3-scale@4"></script>
```
Comment (anthonydillon) [inline templates/openstack/index.html:855]: Do you need d3 here now?

**Review by anthonydillon — APPROVED** [review]
LGTM :+1: with a question about the use of D3

File: templates/openstack/index.html (hunk 1 above)
Comment (carkod) [reply templates/openstack/index.html]: Nope. Thanks. Updated.

**Pull request merged**